### PR TITLE
jwt authentication for doctor login

### PR DESCRIPTION
**Pull request by robsdagreat**

### CODE REVIEW

File: middlewares/doctorAuthentication.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken'
+import CreateDoctorModel from "../models/doctormodel.js";
+
+const authenticateDoctor = async (req, res, next) => {
+ 
+
+  try {
+   
+    const decoded = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET);
```
Comment (renepromesse) [inline middlewares/doctorAuthentication.js:14]: @robsdagreat I see on this line you  used the `token` variable, where did you define that variable? your codes will raise this error, so you have to fix this.
As a hint, the token should come from the request headers, so search about that and fix it

File: middlewares/doctorAuthentication.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const userId = decoded.id;
+
+    const check= await CreateDoctorModel.findById({_id : userId})
+      if (!check) {
+        return res.status(401).json({ error: 'Authentication failed' });
+      }
+
+      req.user = check;
```
Comment (renepromesse) [inline middlewares/doctorAuthentication.js]: here why do you need to set the req.user to check?

**Review by renepromesse — CHANGES_REQUESTED** [review]
Hey @robsdagreat, this is a fabulous work, congratulations.

now look for the feedback and work on it!

File: middlewares/doctorAuthentication.js (hunk 1 above)
Comment (robsdagreat) [reply middlewares/doctorAuthentication.js:14]: Copy that sir, I thought that the headers are sent by the one who makes the login endpoint. I'm gong to change it asap

File: middlewares/doctorAuthentication.js (hunk 2 above)
Comment (robsdagreat) [reply middlewares/doctorAuthentication.js]: I learnt that it is a good practice to attach it on the request for future use, but I'm going to change it as well

**Pull request merged**